docker/prometheus/prometheus-openstack-exporter/Dockerfile.j2

```
@@ -3,17 +3,40 @@ FROM {{ namespace }}/{{ infra_image_prefix }}prometheus-base:{{ tag }}
 LABEL maintainer="{{ maintainer }}" name="{{ image_name }}" build-date="{{ build_date }}"
 {% endblock %}
 
+{% import "macros.j2" as macros with context %}
+
 {% block prometheus_openstack_exporter_header %}{% endblock %}
 
-{% block prometheus_openstack_exporter_repository_version %}
-ENV prometheus_openstack_exporter_version=1.3.0
-{% endblock %}
+{% if base_package_type == 'rpm' %}
+    {% set prometheus_openstack_exporter_packages = [
+        'git',
+        'go',
+        'make',
+    ] %}
+{% elif base_package_type == 'deb' %}
+    {% set prometheus_openstack_exporter_packages = [
+        'build-essential',
+        'git',
+        'golang-go',
+    ] %}
+{% endif %}
+
+{{ macros.install_packages(prometheus_openstack_exporter_packages | customizable("packages")) }}
 
 {% block prometheus_openstack_exporter_install %}
-RUN curl -o /tmp/prometheus_openstack_exporter.tar.gz https://github.com/openstack-exporter/openstack-exporter/releases/download/v${prometheus_openstack_exporter_version}/openstack-exporter-${prometheus_openstack_exporter_version}.linux-{{debian_arch}}.tar.gz \
-    && tar xvf /tmp/prometheus_openstack_exporter.tar.gz -C /opt/ \
-    && rm -f /tmp/prometheus_openstack_exporter.tar.gz \
-    && ln -s /opt/openstack-exporter* /opt/openstack-exporter
+ARG prometheus_openstack_exporter_url=https://github.com/stackhpc/openstack-exporter/archive/refs/heads
```

```suggestion
ARG prometheus_openstack_exporter_url=https://github.com/openstack-exporter/openstack-exporter/archive/refs/heads
```
Maybe default to this instead?

---

It probably makes sense to do everything either in config or code. It might be possible to get all of these changes into a block override, but probably not worth it. i.e. stay with this approach until the patch is merged.